File: src/main/java/ac/dnd/server/common/error/BusinessException.java
package ac.dnd.server.common.error;

import ac.dnd.server.common.support.ErrorCode;

public class BusinessException extends RuntimeException {

	private ErrorCode errorCode;

	public BusinessException(String message, ErrorCode errorCode) {
		super(message);
		this.errorCode = errorCode;
	}

	public BusinessException(ErrorCode errorCode) {
		super(errorCode.getMessage());
		this.errorCode = errorCode;
	}

	public ErrorCode getErrorCode() {
		return errorCode;
	}

}

File: src/main/kotlin/ac/dnd/server/common/error/BusinessException.kt
package ac.dnd.server.common.error

import ac.dnd.server.common.support.ErrorCode

open class BusinessException : RuntimeException {
    val errorCode: ErrorCode

    constructor(message: String, errorCode: ErrorCode) : super(message) {
        this.errorCode = errorCode
    }

    constructor(errorCode: ErrorCode) : super(errorCode.message) {
        this.errorCode = errorCode
    }
}